tests/integrations/test_dsm.py

```diff
@@ -432,6 +432,149 @@ def test_dsmcontext_extraction_base64(self):
         )
 
 
+@features.datastreams_monitoring_support_for_manual_checkpoints
+@scenarios.integrations
+class Test_Dsm_Manual_Checkpoint_Intra_Process:
+    """ Verify DSM stats points for manual checkpoints within the same process thread """
+
+    def setup_dsm_manual_checkpoint_intra_process(self):
+        self.produce = weblog.get(
+            f"/dsm/manual/produce?type=dd-streams&target=system-tests-queue", timeout=DSM_REQUEST_TIMEOUT,
+        )
+        self.consume = weblog.get(
+            f"/dsm/manual/consume?type=dd-streams&source=system-tests-queue&headers={self.produce.text}",
+            timeout=DSM_REQUEST_TIMEOUT,
+        )
+
+    def test_dsm_manual_checkpoint_intra_process(self):
```

Comment:
I believe what we want to test here is checkpointing *without* going through a queue

Reply:
if you look at the implemented endpoints, there is no "queue" the message is going through, its just setting a manual consume/produce checkpoint.